presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/GlueHiveMetastore.java

```diff
@@ -166,7 +168,12 @@ else if (config.getPinGlueClientToCurrentRegion()) {
             }
         }
 
-        if (config.getIamRole().isPresent()) {
+        if (config.getAwsAccessKey().isPresent() && config.getAwsSecretKey().isPresent()) {
```

Comment:
I wonder if `io.prestosql.plugin.hive.s3.PrestoS3FileSystem#createAwsCredentialsProvider` could be extracted and generalized to have common way of providing credentials to S3 and Glue